File: Templates/Project Templates/macOS/System Extension/Network Extension.xctemplate/FilterPacketProvider.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

@implementation ___FILEBASENAME___

- (void)startFilterWithCompletionHandler:(void (^)(NSError *error))completionHandler {

	self.packetHandler = ^NEFilterPacketProviderVerdict(NEFilterPacketContext * _Nonnull context, nw_interface_t  _Nonnull interface, NETrafficDirection direction, const void * _Nonnull packetBytes, const size_t packetLength) {
		return NEFilterPacketProviderVerdictAllow;
	};
}

- (void)stopFilterWithReason:(NEProviderStopReason)reason completionHandler:(void (^)(void))completionHandler {
    // Add code to clean up filter resources.
    completionHandler();
}

@end

File: Templates/Project Templates/macOS/System Extension/Network Extension.xctemplate/FilterPacketProvider.swift
//___FILEHEADER___

import NetworkExtension

class ___FILEBASENAME___: NEFilterPacketProvider {

    override func startFilter(completionHandler: @escaping (Error?) -> Void) {

		packetHandler = { (context, interface, direction, packetBytes, packetLength) in
			return .allow
		}
		completionHandler(nil)
    }
    
    override func stopFilter(with reason: NEProviderStopReason, completionHandler: @escaping () -> Void) {

		// Add code here to tear down the filter
        completionHandler()
    }
}
